src/functional/ffront/foast_to_itir.py

```diff
@@ -31,61 +32,80 @@
 from functional.iterator import ir as itir
 
 
-class TypeKind(enum.Enum):
-    FIELD = 0
-    SCALAR = 1
-    UNKNOWN = 2
+def is_local_kind(symbol_type: ct.FieldType) -> bool:
+    assert isinstance(symbol_type, ct.FieldType)
+    if symbol_type.dims == ...:
+        return False
+    return any(dim.kind == DimensionKind.LOCAL for dim in symbol_type.dims)
 
 
-def resulting_type_kind(symbol_type: ct.SymbolType) -> TypeKind:
-    """
-    Determine whether the resulting type kind of ``symbol_type`` is scalar, field or unknown.
-
-    Useful for determining whether an expression of ``symbol_type`` should be treated as a
-    value or iterator expression during lowering.
-
-    Examples:
-    ---------
-    >>> resulting_type_kind(ct.DeferredSymbolType(constraint=None)).name
-    'UNKNOWN'
+class IteratorTypeKind(enum.Enum):
+    VALUE = 0
+    ITERATOR = 1
+    ENCAPSULATED_ITERATOR = 2
```

Comment:
If you look at it through the eyes of someone unfamiliar with the codebase and the concepts behind it, do you see the double meaning of `iterator` here as problematic?

In `IteratorTypeKind` it means the type kind seen from the iterator level context, in `.ITERATOR` it means the type kind named iterator. That is, if I have understood it correctly.

Reply:
Your understanding is correct. Not sure it is problematic, but it is certainly not optimal so I am open for alternative names.

Reply:
What do you think of `IterViewTypeKind` or something that references "IteratorView" or "ITIR" or "IteratorLevel" as opposed to "Iterator"?

Reply:
I like ITIRTypeKind. Renamed it.